### PR TITLE
Add support for fmod on MPS

### LINKED ISSUE #86810: [MPS] Add support for aten::fmod.Tensor_out for MPS backend
### 🐛 Describe the bug

First time contributors are welcome! 🙂
Add support for [aten::fmod.Tensor_out](https://pytorch.org/docs/stable/generated/torch.fmod.html) for MPS backend. Generic support for adding operations to MPS backend is captured here: https://github.com/pytorch/pytorch/wiki/MPS-Backend#adding-op-for-mps-backend

### Versions

N/A

cc @kulinseth @albanD @malfet @razarmehr @abhudev

### PR DESCRIPTION
**Pull request by pumetu**
Fixes [#86810](https://github.com/pytorch/pytorch/issues/86810)
`fmod` now supported on MPS

### CODE REVIEW

**Comment by PumeTu** [comment]
I replicated how [`std::fmod`](https://en.cppreference.com/w/cpp/numeric/math/fmod) was implemented in C++ for this functionality.

**Comment by kulinseth** [comment]
We don't have support for i64 support for floor op:

```
 (null) Function floorOp_i64 was not found in the library'
```
@PumeTu , can you please raise an error for i64 type not being supported for the `fmod` op.

File: aten/src/ATen/native/mps/operations/BinaryOps.mm [hunk 1]
```diff
@@ -184,6 +184,37 @@ void div_mode_template(const Tensor& self, const Tensor& other,
...
+void fmod_template(const Tensor& self, const Tensor& other,
+                   const Tensor& output, const string op_name)
+{
+  BinaryOpBlock fmod_op_block = ^BinaryOpFn(cachedGraph, primaryCastTensor, secondaryCastTensor) {
+    MPSGraph* mpsGraph = cachedGraph->graph();
+    MPSGraphTensor* absPrimaryTensor = [mpsGraph absoluteWithTensor:primaryCastTensor
+                                                               name:nil];
+    MPSGraphTensor* absSecondaryTensor = [mpsGraph absoluteWithTensor:secondaryCastTensor
+                                                                 name:nil];
+    MPSGraphTensor* modTensor = [mpsGraph floorModuloWithPrimaryTensor:absPrimaryTensor
```
Comment (kulinseth) [inline aten/src/ATen/native/mps/operations/BinaryOps.mm:196]: This doesn't have support for i64 as mentioned below. Can you raise a TORCH_CHECK error for it?

**Comment by PumeTu** [comment]
@kulinseth Added a check for i64 in `fmod`.

**Review by kulinseth — APPROVED** [review]
Looks good.

**Comment by PumeTu** [comment]
@kulinseth Are you able to force merge the issue as CUDA errors are unrelated to MPS?

**Comment by YankoFelipe** [comment]
https://github.com/pytorch/pytorch/actions/runs/3490499797/jobs/5848016852#logs
@kulinseth @PumeTu can this failure be fixed so that the code can be merged?

**Comment by kulinseth** [comment]
@PumeTu , thanks for this PR. We ended up enabling the Op to make it to PyTorch 2.0 release. https://github.com/pytorch/pytorch/pull/94722

**Pull request closed without merging**